swift/Tests/Blockchains/THORSwapTests.swift

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import WalletCore
+
+class THORSwapTests: XCTestCase {
+
+    func testSignerEthBnb() {
+        // prepare swap input
+        let input = THORSwapSwapInput.with {
+            $0.fromChain = .eth
+            $0.fromAddress = "0xb9f5771c27664bf2282d98e09d7f50cec7cb01a7"
+            $0.toAsset = THORSwapAsset.with {
+                $0.chain = .bnb
+                $0.symbol = "BNB"
+                $0.tokenID = ""
+            }
+            $0.toAddress = "bnb1us47wdhfx08ch97zdueh3x3u5murfrx30jecrx"
+            $0.vaultAddress = "0x1091c4De6a3cF09CdA00AbDAeD42c7c3B69C83EC"
+            $0.routerAddress = "0x42A5Ed456650a09Dc10EBc6361A7480fDd61f27B"
+            $0.fromAmount = "50000000000000000"
+            $0.toAmountLimit = "600003"
+        }
+
+        // serialize input
+        let inputSerialized = try? input.serializedData()
```

Comment:
You can add `throws` to `testSignerEthBnb`, so that you don't need to force wrap them every time (see other swift tests)